tests/ha/control_pod/test_control_pod_soft_failures.py

```diff
@@ -0,0 +1,617 @@
+                LOGGER.info("Step 2.2: Failed IAM user creation as expected")
+            else:
+                assert_utils.assert_not_equal(user, None, "Expected successful IAM user creation")
+                self.s3_clean.update(user)
+                self.uids.append(user[list(user.keys())[0]]["user_name"])
+                LOGGER.info("Step 2.2: Successfully created IAM user")
+            d_user = del_users.pop(list(del_users.keys())[0])
+            LOGGER.info("Step 2.3: Delete iam user %s", d_user["user_name"])
+            d_user_dict = {d_user["user_name"]: d_user}
+            resp = self.ha_obj.delete_iam_user_with_header(user=d_user_dict, header=self.header)
+            if pod == self.ctrl_pod_list[-1]:
+                assert_utils.assert_false(resp[0], "Expected failure in IAM user deletion")
+                LOGGER.info("Step 2.3: Failed IAM user deletion as expected")
+                self.uids.append(d_user["user_name"])
+                self.s3_clean.update(d_user_dict)
+            else:
+                assert_utils.assert_true(resp[0], "Expected successful IAM user deletion")
+                LOGGER.info("Step 2.3: Successfully deleted IAM user")
+
+            LOGGER.info("Step 2.4: Verify if %s IAM users are persistent across control pod "
+                        "soft failures", self.uids)
+            for user in self.uids:
+                resp = self.rest_iam_user.get_iam_user(user)
+                assert_utils.assert_equal(int(resp.status_code), HTTPStatus.OK.value,
+                                          f"Couldn't find user {user} after control pod "
+                                          "failover")
+                LOGGER.info("Step 2.4: User %s is persistent: %s", user, resp)
+        LOGGER.info("Step 2: Created soft failures and performed IAM user creation")
+
+        LOGGER.info("Step 3: Checking cluster status")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        assert_utils.assert_false(resp[0], resp[1])
+        LOGGER.info("Step 3: Cluster status should be bad as control pods are in failed state")
+
+        LOGGER.info("Step 4: Perform WRITE-READ-Verify on user %s", self.uids[0])
+        self.test_prefix = 'test-45499-new'
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(self.users.values())[0],
+                                                    log_prefix=self.test_prefix,
+                                                    skipcleanup=True, nsamples=5, nclients=5)
+        assert_utils.assert_true(resp[0], resp[1])
+        LOGGER.info("Step 4: Performed WRITEs-READs-Verify with variable sizes objects.")
+
+        LOGGER.info("Waiting for %s time for all control pod containers to start",
+                    self.max_proc_restart_delay)
+        time.sleep(self.max_proc_restart_delay)
+
+        LOGGER.info("Step 5: Checking cluster status")
+        resp = self.ha_obj.poll_cluster_status(self.node_master_list[0], timeout=180)
```

Comment:
why timeout again? it can very well be same as delay?

Reply:
So even if service is up, it may take time to reflect in status check. Just like we do for server and data pods.